toolchain/check/testdata/function/declaration/no_prelude/name_poisoning.carbon

```diff
@@ -311,10 +311,21 @@ interface A {
 class X {
   extend impl as A {
     fn F() { return; }
-    // CHECK:STDERR: fail_using_poisoned_name_in_impl_outside_class.carbon:[[@LINE+4]]:10: error: `impl as` can only be used in a class [ImplAsOutsideClass]
+    // CHECK:STDERR: fail_using_poisoned_name_in_impl_outside_class.carbon:[[@LINE+15]]:10: error: `impl as` can only be used in a class [ImplAsOutsideClass]
     // CHECK:STDERR:     impl as B {}
     // CHECK:STDERR:          ^~
     // CHECK:STDERR:
+    // CHECK:STDERR: fail_using_poisoned_name_in_impl_outside_class.carbon:[[@LINE+11]]:13: error: `Core.ImplicitAs` implicitly referenced here, but package `Core` not found [CoreNotFound]
+    // CHECK:STDERR:     impl as B {}
+    // CHECK:STDERR:             ^
+    // CHECK:STDERR:
```

Comment:
It would be nice to suppress this error given that an error has already been issued.

Reply:
Once the fn is named B, the error changes to trying to do impl definition on `fn B` and failing. I think we want that error, it's kind of the point of the change that we keep typechecking. I don't know how to suppress the error (coming from convert.cpp), without just not creating the impl at all, but then we fail to typecheck everything else in the impl and that seems worse?

Reply:
If we are writing an `impl as` in a context where that is nonsense, I don't think we want to diagnose more than once saying it is nonsense, no matter what else is in that declaration or body of that `impl as`. I do think continuing to check the enclosing scope is good.

Reply:
Unfortunately, I don't think we can stop checking other stuff inside the impl, those are spread over all the rest of check and they'd all have to go look if we're inside an `impl as` and skipping checking. Maybe you know some way to skip checking a subtree that I am unaware of?

That said, I made the test show we continue checking by adding a fn call into the body of B() and placing it after the invalid `impl as` so that we can clearly see it in the semir output. Also did that for the other tests in the file.